src/include/sof/audio/buffer.h

```diff
@@ -35,6 +35,9 @@ struct comp_dev;
 #define BUFF_CB_TYPE_PRODUCE	BIT(0)
 #define BUFF_CB_TYPE_CONSUME	BIT(1)
 
+/* dma buffer hw required alignment*/
+#define DMA_BUFFER_ALIGNMENT	128
```

Comment:
It definitely shouldn't be here. It's HD-DMA specific. The best solution would be to fetch DMA_BUFFER_ADDRESS_ALIGNMENT as DMA attribute in host_params (host.c) and realloc the buffer if needed. Right now you are aligning every possible DMA buffer to 128 bytes.

Reply:
By `attribute` do you mean something that should come in ipc/caps/params or f.e. `hda_dma_get_attribute`? IMHO the 2nd one is better as FW should know what alignment is valid, driver shouldn't care  about it.

Reply:
`hda_dma_get_attribute`

Reply:
@tlauda I did change as per request however we still have glitches now after realloc. Could You please take a look in a spare time? I will be debugging it today.